src/advuilist.h

```diff
@@ -0,0 +1,889 @@
+        std::size_t _pagesize = 0;
+        list_t _list;
+        colscont_t _columns;
+        sortcont_t _sorters;
+        groupercont_t _groupers;
+        groupcont_t _groups;
+        pagecont_t _pages;
+        ffilter_t _ffilter;
+        fcounter_t _fcounter;
+        frebuild_t _frebuild;
+        fdraw_t _fdraw;
+        fctxt_t _fctxt;
+        std::string _filter;
+        std::string _filterdsc;
+        typename sortcont_t::size_type _csort = 0;
+        typename groupercont_t::size_type _cgroup = 0;
+        typename list_t::size_type _cidx = 0;
+        typename pagecont_t::size_type _cpage = 0;
+        /// total column width weights
+        cwidth_t _tweight = 0;
+        int _innerw = 0;
+        bool _exit = true;
+
+        input_context _ctxt;
+        catacurses::window _w;
+        std::shared_ptr<ui_adaptor> _ui;
+
+        Container *_olist = nullptr;
+
+        /// number of lines at the top of the window reserved for decorations
+        static constexpr int const _headersize = 2;
+        /// number of lines at the bottom of the window reserved for decorations
+        static constexpr int const _footersize = 1;
+        /// first usable column after decorations
+        static constexpr int const _firstcol = 1;
+        /// minimum whitespace between columns
+        static constexpr int const _colspacing = 1;
+
+        select_t _peek( std::size_t amount );
+        select_t _peekall();
+        std::size_t _count( std::size_t idx );
+        std::size_t _peekcount();
+
+        void _initui();
+        void _initctxt();
+        void _print();
+        int _printcol( col_t const &col, std::string const &str, point const &p, nc_color const &color );
+        void _printheaders();
+        void _printfooters();
+        void _sort( typename sortcont_t::size_type idx );
+        void _group( typename groupercont_t::size_type idx );
+        void _querysort();
+        void _queryfilter();
+        std::size_t _querypartial( std::size_t max );
+        void _setfilter( std::string const &filter );
+        bool _basicfilter( T const &it, std::string const &filter ) const;
+        typename pagecont_t::size_type _idxtopage( typename list_t::size_type idx );
+        void _incidx( std::size_t amount );
+        void _decidx( std::size_t amount );
+        void _setidx( std::size_t idx );
+};
+
+// *INDENT-OFF*
+
+#ifdef __clang__
+#pragma clang diagnostic push
+// travis' old clang wants a change that breaks compilation with newer versions
+#pragma clang diagnostic ignored "-Wmissing-braces"
+#endif
+
+template <class Container, typename T>
+advuilist<Container, T>::advuilist( Container *list, point size, point origin,
+                                    std::string const &ctxtname, bool init )
+    : _size( size.x > 0 ? size.x : TERMX / 4, size.y > 0 ? size.y : TERMY / 4 ),
+      _origin( origin.x >= 0 ? origin.x : TERMX / 2 - _size.x / 2,
+               origin.y >= 0 ? origin.y : TERMY / 2 - _size.y / 2 ),
```

Comment:
You should always put the logic of window position/size calculation in the resize callback, in case values such as `TERMX`/`TERMY` are changed when the resize callback is called during a resize event.

Reply:
Thanks; I'll see what I can do

Reply:
I've taken a look at this and I'm not sure if there's anything I can do unless I limit these UI elements to a fixed screen position. I'm only using `TERMX`/`TERMY` if a size/position isn't specified in the constructor. Is there some heuristic I can do to reposition the window in the general case? Are there many cases where I need to resize the windows? I can only think of an android user rotating his device. I can probably write a failsafe though that tries to fit the window on the new screen without considering initial size and position.

Reply:
I think you can save `size` and `origin` in the class and calculate the actual size on the fly in the resize callback.

Reply:
I've implemented resizing support in 4ef8918 but I'm not very happy with it. Please take a look.